### PR TITLE
Supporting > 10k federal awards

### PR DESCRIPTION
**Pull request by phildominguez-gsa**
Addresses https://github.com/GSA-TTS/FAC/issues/3529

In this PR:
- Updating `build_templates` to look for `MAX_ROWS` env var, this way we can create workbooks with 20k rows
- We've determined that uploading workbooks > 10k rows only has issues with federal award # entries being greater than 4 digits. Because of this, we're updating regexes/schemas to allow federal award #s to have 4 or 5 digits, with the plan to only allow 5 at some point down the road.
- Adding a bunch of workbooks to `/should_pass` and `/should_fail` to cover these cases
- Increasing upload timeout to 4m, as it took ~3m to upload the 20k line workbook in preview
- It seems `make build_templates` wasn't run after changing an error message for `FederalProgramNameValidation`, so I guess that's being pulled in too

Testing:
- Generating workbooks
  - cd into `/schemas`
  - `make build_templates` should still create a 10k row workbook
  - `MAX_ROWS=100 make build_templates` should create a 100 row workbook
  - `MAX_ROWS=asdf make build_templates` should still create a 10k row workbook
- Upload testing
  - Start a submission
  - Normal federal awards and audit findings uploads should work (the ones in `audit/fixtures/workbooks/should_pass/134732-21/` are sufficient)
  - Uploading a [20k federal awards workbook](https://github.com/GSA-TTS/FAC/files/14723167/Copy.of.federal-awards-workbook-lots-of-rows.xlsx) should work without timing out
  - Modify one of the federal award numbers to have 5 digits (such as `AWARD-0041` to `AWARD-00410`). Both uploads should fail, since all must have the same length.
  - Modify all federal award numbers to have 5 digits. Uploading these should validate.
  - Upload federal award and audit findings workbooks such that one of them uses 4 digit numbers, and the other uses 5 digits. Cross validation should fail here. Uploading ones with both of length 4 or both of length 5 should pass.

## PR checklist: submitters

- [ ]   Link to an issue if possible. If there’s no issue, describe what your branch does. Even if there is an issue, a brief description in the PR is still useful.
- [ ]   List any special steps reviewers have to follow to test the PR. For example, adding a local environment variable, creating a local test file, etc.
- [ ]   For extra credit, submit a screen recording like [this one](https://github.com/GSA-TTS/FAC/pull/1821).
- [ ]   Make sure you’ve merged `main` into your branch shortly before creating the PR. (You should also be merging `main` into your branch regularly during development.)
- [ ]   Make sure you’ve accounted for any migrations. When you’re about to create the PR, bring up the application locally and then run `git status | grep migrations`. If there are any results, you probably need to add them to the branch for the PR. Your PR should have only **one** new migration file for each of the component apps, except in rare circumstances; you may need to delete some and re-run `python manage.py makemigrations` to reduce the number to one. (Also, unless in exceptional circumstances, your PR should not delete any migration files.)
- [ ]   Make sure that whatever feature you’re adding has tests that cover the feature. This includes test coverage to make sure that the previous workflow still works, if applicable.
- [ ]   Make sure the full-submission.cy.js [Cypress test](https://github.com/GSA-TTS/FAC/blob/main/docs/testing.md#end-to-end-testing) passes, if applicable.
- [ ]   Do manual testing locally. Our tests are not good enough yet to allow us to skip this step. If that’s not applicable for some reason, check this box.
- [ ]   Verify that no Git surgery was necessary, or, if it was necessary at any point, repeat the testing after it’s finished.
- [ ]   Once a PR is merged, keep an eye on it until it’s deployed to dev, and do enough testing on dev to verify that it deployed successfully, the feature works as expected, and the happy path for the broad feature area (such as submission) still works.

## PR checklist: reviewers

- [ ]   Pull the branch to your local environment and run `make docker-clean; make docker-first-run && docker compose up`; then run `docker compose exec web /bin/bash -c "python manage.py test"`
- [ ]   Manually test out the changes locally, or check this box to verify that it wasn’t applicable in this case.
- [ ]   Check that the PR has appropriate tests. Look out for changes in HTML/JS/JSON Schema logic that may need to be captured in Python tests even though the logic isn’t in Python.
- [ ]   Verify that no Git surgery is necessary at any point (such as during a merge party), or, if it was, repeat the testing after it’s finished.

The larger the PR, the stricter we should be about these points.

### CODE REVIEW

File: backend/schemas/source/excel/libs/SheetValidations.libsonnet [hunk 1]
```diff
@@ -86,7 +86,7 @@ local LoanBalanceValidation = {
 
 local AwardReferenceValidation = {
   type: 'custom',
-  formula1: '=AND(LEN(FIRSTCELLREF)=10, LEFT(FIRSTCELLREF, 6)="AWARD-", ISNUMBER(VALUE(MID(FIRSTCELLREF, 7, 4))), NOT(FIRSTCELLREF="AWARD-0000"))',
+  formula1: '=AND(LEN(FIRSTCELLREF)=11, LEFT(FIRSTCELLREF, 6)="AWARD-", ISNUMBER(VALUE(MID(FIRSTCELLREF, 7, 5))), NOT(FIRSTCELLREF="AWARD-00000"))',
   custom_error: 'The value should follow the pattern AWARD-#### (where #### is a four-digit number).',
```
Comment (tadhg-ohiggins) [inline backend/schemas/source/excel/libs/SheetValidations.libsonnet:90]:
```suggestion
  custom_error: 'The value should follow the pattern AWARD-##### (where ##### is a five-digit number).',
```

Comment (tadhg-ohiggins) [reply backend/schemas/source/excel/libs/SheetValidations.libsonnet:90]: Given this change, we should also update the workbook version number, and the list of versions we accept. @phildominguez-gsa 

Per @jadudm the new version number should be `1.1.0`.

The workbook version number is in `excel/libs/sheets.libsonnet`. 

The list of versions we accept is in `audit/intakelib/checks/check_version_number.py`.

We should modify the `should_pass` workbooks with length five award identifiers to be version `1.1.0` to exercise the `check_version_number.py` change.

@jadudm is unsure that this is an exhaustive list of version-related changes.

**Review by tadhg-ohiggins — CHANGES_REQUESTED** [review]

File: backend/schemas/source/excel/libs/SheetValidations.libsonnet (hunk 1 above)
Comment (jperson1) [reply backend/schemas/source/excel/libs/SheetValidations.libsonnet]: @phildominguez-gsa The output schema json should also be regenerated to include the the 1.1.0 version change - that falls under the `build_sections` portion of the Makefile

Comment (phildominguez-gsa) [reply backend/schemas/source/excel/libs/SheetValidations.libsonnet]: Done, and tests pass

**Review by jadudm — APPROVED** [review]
LGTM. Tested in preview, timeouts confirmed. Tests pass, workbooks with data upload in preview. Reviewed with Tadhg and Phil.

File: backend/schemas/source/excel/libs/SheetValidations.libsonnet (hunk 1 above)
Comment (jadudm) [reply backend/schemas/source/excel/libs/SheetValidations.libsonnet]: Resolved through edits and reviewed by @jadudm , @phildominguez-gsa , and @tadhg-ohiggins .

**Review by tadhg-ohiggins — APPROVED** [review]

**Pull request merged**